python/dolfinx/fem/assemble.py

```diff
@@ -183,7 +184,7 @@ def _assemble_vector_form(L: Form, constants=None, coeffs=None) -> la.Vector:
 
 
 @assemble_vector.register(np.ndarray)
-def _assemble_vector_array(b: np.ndarray, L: Form, constants=None, coeffs=None):
+def _assemble_vector_array(b: npt.NDArray, L: Form, constants=None, coeffs=None):
```

Comment:
```suggestion
def _assemble_vector_array(b: npt.NDArray, L: Form, constants=None, coeffs=None):
```
I think we should revert this one.

Rationale:
`npt.NDArray` is neat when you know the `dtype` of your array (for instance int32, int64, float64, float32 etc as you can then use `npt.NDArray[np.int32]`.

Howver, for the assembly operators, `b` can take multiple dtypes, making it more natural to let it be kept as a generic `np.ndarray` with no datatype specification.